app/assets/javascripts/shared/editor_toolbar.js

```diff
@@ -72,60 +71,48 @@ class EditorToolbar {
       }
     });
 
-    // Handler for setting the correct textarea height on keyboard input
-    this.$target[0].addEventListener('input', setHeight);
+    // Handler for setting the correct textarea height on keyboard input, when
+    // focus is lost, or when content is inserted programmatically
+    // Handler for setting the correct textarea height when focus is lost, or
+    // when content is inserted programmatically
+    this.$target.on('blur textchange input', this.setHeight);
 
-    function setHeight(e) {
-      const shrinkEvents = ['deleteContentForward', 'deleteContentBackward', 'deleteByCut', 'historyUndo', 'historyRedo'];
-
-      if (shrinkEvents.includes(e.inputType)) {
-        // shrink the text area when content is being removed
-        $(this).css({'height': '1px'});
-      }
-      
-      // expand the textarea to fix the content
-      $(this).css({'height': this.scrollHeight + 2});
-    };
-  
     // Handler for setting the correct textarea heights on load (for current values)
-    this.$target.each(setHeight);
-
-    // Handler for setting the correct textarea height when focus is lost
-    this.$target.on('blur', setHeight);
+    this.$target.each(this.setHeight);
 
     // when a toolbar button is clicked
     this.$editorToolbar.find('[data-btn]').click(function() {
-      var $element = that.$editorField.find('textarea, input[type=text]');
-      var affix = that.affixes[$(this).data('btn')];
-  
       if ($(this).is('[data-btn~=image')) {
         that.$fileField.click();
       } else {
-        // inject markdown
-        that.injectSyntax($element, affix);
+        var cursorInfo = that.$target.cursorInfo(),
+            affix = that.affixesLibrary($(this).data('btn'), cursorInfo.text);
+
+        that.injectSyntax(affix);
       }
     });
 
     // keyboard shortcuts
-    this.$editorField.find('textarea, input[type=text]').keydown(function(e) {
-      var key = e.which || e.keyCode; // for cross-browser compatibility
-
-      if (e.metaKey && key === 66 ) { // 66 = b
-        e.preventDefault();
-        that.$editorToolbar.find('[data-btn~=bold]').click();
-      }
-      else if (e.metaKey && key === 73 ) { // 73 = i
-        e.preventDefault();
-        that.$editorToolbar.find('[data-btn~=italic]').click();
-      }
-      else if (e.metaKey && key === 75 ) { // 75 = k
-        e.preventDefault();
-        that.$editorToolbar.find('[data-btn~=link]').click();
+    this.$target.keydown(function(e) {
+      var key = e.which || e.keyCode, // for cross-browser compatibility
+          selector = undefined;
```

Comment:
do we need to initialize `selector` as `undefined`? doesn't that happen by default?

Reply:
Yeah I think you're correct.